timemachine/cpp/src/summed_potential.cu

```diff
@@ -0,0 +1,47 @@
+#include "summed_potential.hpp"
+#include <stdexcept>
+
+namespace timemachine {
+
+SummedPotential::SummedPotential(std::vector<Potential *> potentials, std::vector<int> params_sizes)
+    : potentials_(potentials), params_sizes_(params_sizes) {
+    if (potentials_.size() != params_sizes_.size()) {
+        throw std::runtime_error("number of potentials != number of parameter sizes");
+    }
+};
+
+void SummedPotential::execute_device(
```

Comment:
Do you know how much faster this is than the previous version?

Reply:
I just ran a timing test (code [here](https://gist.github.com/mcwitt/21e992891eb2e51081fe974743503fef)). It looks like there's a modest ~20% speedup for computing U, and a ~50% speedup for computing the gradient:

```
===============================================================================
construct_differentiable_interface (precision=<class 'numpy.float32'>)
  construct: 12.506331214000966
  evaluate U  (1000 iters): 0.7602241589993355
  evaluate ∇U (1000 iters): 9.319278081000448
===============================================================================
construct_differentiable_interface_fast (precision=<class 'numpy.float32'>)
  construct: 12.518712601999141
  evaluate U  (1000 iters): 0.617156143000102
  evaluate ∇U (1000 iters): 4.233765714001493
```

Reply:
nice!

Reply:
The above timing was only computing dU/dx. I updated to also compute dU/dp and dU/dl ([here](https://gist.github.com/mcwitt/4e65ef5e9356e264d8c450d45f329934)). The relative timing is pretty much unchanged:

```
===============================================================================
construct_differentiable_interface (precision=<class 'numpy.float32'>)
  construct: 12.515916177999316
  evaluate U  (1000 iters): 0.7570085630013637
  evaluate ∇U (1000 iters): 11.697148505998484
===============================================================================
construct_differentiable_interface_fast (precision=<class 'numpy.float32'>)
  construct: 12.499262368000927
  evaluate U  (1000 iters): 0.6442218980009784
  evaluate ∇U (1000 iters): 5.53595667800073
```